### DISCUSSION TITLE
proxmox lcx container mount point

### DISCUSSION THREAD

**Discussion by redpimpernel** [bpg/terraform-provider-proxmox#760] [Q&A] [answered]
Hi, I have just started learning proxmox and I am stuck trying to create a lxc container using this terraform provider.
My issue is the "mount_point" option.
I am using a non-root user for terraform and I cant get it to work and understand how to pass options that would be same as the default if setting it up through the GUI.
Any guidance would be greatly appreciated.

**Comment by bpg-dev** [comment]
Hi @redpimpernel! Could you please share a screenshot from the UI with the mount point setup  you're trying to achieve? I assume you can successfully do it as a _non-root user_ in UI as well?

Some mount point configuration will require you to use root though. For example, the bind mount we have in the [container example template](https://registry.terraform.io/providers/bpg/proxmox/latest/docs/resources/virtual_environment_container). Without root user auth, the provider will throw an error like
```
╷
│ Error: error creating container: received an HTTP 403 response - Reason: Permission check failed (mount point type bind is only allowed for root@pam)
│ 
```

**Comment by redpimpernel** [comment]
Hi,
I am really sorry but, I'm not sure how to provide what you are asking.

Yes, I can create a container through the GUI as my non-root user.  That 403 error is what I am experiencing when following the example.

All I am trying to achieve is, create a container using all the defaults you get when you just click through the GUI changing only what is necessary.

When creating a container, you only need to pass a path if you are attaching a second disk, not for creating the rootfs file system. So I am not sure what I am supposed to put as the "mount_point" option to set up the container the same way.

Thank you for helping.

**Reply by bpg-dev** [reply]
Ah, I see what you mean, basically how to make the same but in terraform template, right?
<img width="805" alt="Screenshot 2023-12-06 at 12 11 02 PM" src="https://github.com/bpg/terraform-provider-proxmox/assets/627562/efc80ac8-b2f4-4a26-8b6f-7f9a808ed6bb">

I'm not using containers much in my lab, so I many of the lxc features were developed by community, and I haven't touched many of them. Mount points is one of such feature 😅.

From reading the code I'd say something like that _should_ work:
```hcl
 mount_point {
    volume = "local-lvm"
    size   = "8G"
    path   = "/mnt/local"
  }
```
... but it does not, at least in my setup, and I don't see a good reason why.

Could you try that as well? 
In a mean time I'll try to debug and see what's going on there.

**Reply by bpg-dev** [reply, marked as answer]
I found a few bugs around volume mount support, will fix them for the next release:
- #772

**Comment by redpimpernel** [comment]
Oh Awesome! Love your work
